src/rdkafka_telemetry_decode.c

```diff
@@ -0,0 +1,570 @@
+        metric.description.funcs.decode = &decode_string;
+        metric.description.arg          = decode_interface;
+        metric.cb_data.funcs.decode     = &data_msg_callback;
+        metric.cb_data.arg              = decode_interface;
+
+        if (!pb_decode(stream, opentelemetry_proto_metrics_v1_Metric_fields,
+                       &metric)) {
+                RD_INTERFACE_CALL(decode_interface, decode_error,
+                                  "Failed to decode Metric: %s",
+                                  PB_GET_ERROR(stream));
+                return false;
+        }
+
+        return true;
+}
+
+static bool decode_scope_metrics(pb_istream_t *stream,
+                                 const pb_field_t *field,
+                                 void **arg) {
+        rd_kafka_telemetry_decode_interface_t *decode_interface = *arg;
+        opentelemetry_proto_metrics_v1_ScopeMetrics scope_metrics =
+            opentelemetry_proto_metrics_v1_ScopeMetrics_init_zero;
+        scope_metrics.scope.name.funcs.decode    = &decode_string;
+        scope_metrics.scope.name.arg             = decode_interface;
+        scope_metrics.scope.version.funcs.decode = &decode_string;
+        scope_metrics.scope.version.arg          = decode_interface;
+        scope_metrics.metrics.funcs.decode       = &decode_metric;
+        scope_metrics.metrics.arg                = decode_interface;
+
+        if (!pb_decode(stream,
+                       opentelemetry_proto_metrics_v1_ScopeMetrics_fields,
+                       &scope_metrics)) {
+                RD_INTERFACE_CALL(decode_interface, decode_error,
+                                  "Failed to decode ScopeMetrics: %s",
+                                  PB_GET_ERROR(stream));
+                return false;
+        }
+        return true;
+}
+
+static bool decode_resource_metrics(pb_istream_t *stream,
+                                    const pb_field_t *field,
+                                    void **arg) {
+        rd_kafka_telemetry_decode_interface_t *decode_interface = *arg;
+        opentelemetry_proto_metrics_v1_ResourceMetrics resource_metrics =
+            opentelemetry_proto_metrics_v1_ResourceMetrics_init_zero;
+        resource_metrics.resource.attributes.funcs.decode = &decode_key_value;
+        resource_metrics.resource.attributes.arg          = decode_interface;
+        resource_metrics.scope_metrics.funcs.decode = &decode_scope_metrics;
+        resource_metrics.scope_metrics.arg          = decode_interface;
+        if (!pb_decode(stream,
+                       opentelemetry_proto_metrics_v1_ResourceMetrics_fields,
+                       &resource_metrics)) {
+                RD_INTERFACE_CALL(decode_interface, decode_error,
+                                  "Failed to decode ResourceMetrics: %s",
+                                  PB_GET_ERROR(stream));
+                return false;
+        }
+        return true;
+}
+
+#ifdef WITH_SNAPPY
```

Comment:
Change this otherwise `WITH_SNAPPY` is always defined even if 0.
```suggestion
#if WITH_SNAPPY
```

Reply:
Updated